waspc/cli/Command/Compile.hs

```diff
@@ -0,0 +1,35 @@
+module Command.Compile
+    ( compile
+    ) where
+
+import           Control.Monad.IO.Class (liftIO)
+import           Data.List              (find, isSuffixOf)
+import qualified Path                   as P
+
+import qualified Common
+import           CompileOptions         (CompileOptions (..))
+import qualified Lib
+import           StrongPath             (Abs, Dir, Path, (</>))
+import qualified StrongPath             as SP
+import qualified Util.IO
+
+
+-- | Compiles Wasp source code in waspRoot directory and generates a project in given outDir directory.
+compile :: Path Abs (Dir Common.WaspProjectDir) -> Path Abs (Dir Lib.ProjectRootDir) -> IO (Either String ())
```

Comment:
so here we have cli/Command/Compile, but compile is not actually a (cli) command? You are rather here doing what I named "operation" earlier. This is the confusing part, I think we should somehow better differentiate that.
I tried using Job -> Operation -> Command, but let's see if we can think of something better.

Reply:
Decided to keep it like this for now and we will see how to best organize it once it grows further.